### PR TITLE
Adding Popup Helper Text to Add Project Update Form

### PR DESCRIPTION
**Pull request by zhongh**
Initial commit. Message box boundary still not working.

### CODE REVIEW

**Comment by zhongh** [comment]
screen shot 2016-02-05 at 7 15 00 pm

**Comment by mrpatrickwest** [comment]
I would like to have helper text for all of the fields on that page.

Can we add black border to the popup?

**Comment by zhongh** [comment]
I will add helper text for all.
The border is the pain right now. Just can't get the border working but I will keep trying...

**Comment by mrpatrickwest** [comment]
Looks like you are not up to date with master. You're missing the instrument reference and the modification label is still there.

**Comment by zednis** [comment]
I used Intelli-J IDE to attempt to resolve merge conflicts, but based on the Travis-CI failure something did not work right.

**Comment by zednis** [comment]
@mrpatrickwest `ant test` of this branch works on my local machine.  I am not sure if the travis-CI failure was not just a network problem...

Could you checkout a clean repo and test the build of this branch?

**Comment by am-e** [comment]
I may have made some additional changes when working on the "refers to instrument" branch.

**Comment by mrpatrickwest** [comment]
@am-e May have made changes to ???

**Comment by mrpatrickwest** [comment]
There's a commit that changes the submodule versions, that's what's breaking

**Comment by zhongh** [comment]
@mrpatrickwest While making some changes to instruments, @am-e also made a little update to the publications part above it. After merging there emerged a redundant line. The commit just now was to fix that conflict.

And how to fix the submodule version problem?

**Comment by mrpatrickwest** [comment]
@zednis I did a fresh clone and checkout of the branch on my laptop
git submodule init - works fine
git submodule update - fails for vitro

**Comment by zhongh** [comment]
Here is how I fixed it:
`git submodule deinit -f .`
`git checkout 183512fd148c935a9dcf83b5aabd99224f203c21 -- vitro`
`git submodule init`
`git submodule update`

**Comment by am-e** [comment]
I did make changes but they were all limited to the addProjectUpdateGenerator.java and the ftl, (also the preprocessor), took out some unused variables and added error checking for missing values.

File: productMods/templates/freemarker/edit/forms/addProjectUpdate.ftl [hunk 1]
```diff
@@ -180,6 +185,39 @@ Set this flag on the input acUriReceiver where you would like this behavior to o
...
+        position:absolute;
+        width: 20em;
+        height: 6em;
+        bottom: 1em;
+        left: 1em;
+        z-index:1000;
+        background-color:#FFFFFF;
+        padding: 5px;
+        border-radius: 4px;
+        border-top-color: #5C85D6;
```
Comment (mrpatrickwest) [inline productMods/templates/freemarker/edit/forms/addProjectUpdate.ftl:242]: these border styles were already defined by parent styles so needed to override them.

**Comment by mrpatrickwest** [comment]
need to add helper text for all of the other fields as well. Just add placeholder text in there for now and I'll work with Josh on some good text to put there.

**Comment by zhonghao1014** [comment]
All initial placeholder helper texts created for further discussion between Patrick and Josh. Border style overriding for popup box still not working but I will keep trying.
image

**Comment by zhongh** [comment]
1) Borders working (Thanks @mrpatrickwest !!)
2) All initial helper texts added for Patrick and Josh to review. 
3) Managed to did a git pull on the local branch on udco, but failed to compile due to lack of sudoer permission.
image

File: productMods/templates/freemarker/edit/forms/addProjectUpdate.ftl (hunk 1 above)
Comment (zhongh) [inline productMods/templates/freemarker/edit/forms/addProjectUpdate.ftl:242]: :+1:

**Comment by mrpatrickwest** [comment]
@zednis @zhonghao1014 @am-e 
Ready to be reviewed. Login, find a project, click the Update tab, click the + to add a project update.

**Comment by am-e** [comment]
Logged in, checked the help texts, created project update, all seems to be working.

**Comment by zhongh** [comment]
:+1: for me

**Comment by zednis** [comment]
:+1:

**Pull request merged**